src/toil/leader.py

```diff
@@ -753,8 +753,14 @@ def issueJob(self, jobNode):
             workerCommand.append(base64.b64encode(pickle.dumps(context)).decode('utf-8'))
 
         jobNode.command = ' '.join(workerCommand)
+
+        job_environment = {
+            # set the correct number of cores used by OpenMP applications
+            'OMP_NUM_THREADS': str(jobNode.cores),
```

Comment:
Maybe we should add this to `environment.rst` with a note about how we provide it to jobs ourselves, instead of letting the user pass it through with the normal command line options that work on other environment variables?

Reply:
I modified this so that it'll use the `OMP_NUM_THREADS` variable if it is set in the environment, otherwise it'll set it for each job based on `job.cores`.